files/static/index.js

```diff
@@ -0,0 +1,340 @@
+      }
+      break;
+    case "removed":
+      if (checkDomainIssues.warnings.length === 0) {
+        $("#status").textContent = "" + domain + " was previously on the preload list, but has been removed. It may be submitted again.";
+      } else {
+        $("#status").textContent = "" + domain + " was previously on the preload list, but has been removed.";
+      }
+      break;
+    default:
+      $("#status").textContent = "Cannot determine preload status.";
+    }
+
+    function createIssueElement(issue, bullet, className) {
+      var el = document.createElement("div");
+      el.classList.add(className);
+      // summary.title = issue.code;
+
+      var bulletSpan = document.createElement("span");
+      bulletSpan.textContent = bullet + " ";
+      el.appendChild(bulletSpan);
+
+      var summary = document.createElement("span")
+      summary.classList.add("summary");
+      summary.textContent = issue.summary;
+      el.appendChild(summary);
+
+      var br = document.createElement("br")
+      el.appendChild(br);
+
+      var message = document.createElement("span")
+      message.textContent = issue.message;
+      el.appendChild(message);
+
+      return el;
+    }
+
+    $("#errors").textContent = "";
+    $("#warnings").textContent = "";
+    for (var i in checkDomainIssues.errors) {
+      var error = checkDomainIssues.errors[i];
+      var el = createIssueElement(error, "❌", "error");
+      $("#errors").appendChild(el);
+    }
+    for (var j in checkDomainIssues.warnings) {
+      var warning = checkDomainIssues.warnings[j];
+      var el = createIssueElement(warning, "⚠️", "warning");
+      $("#warnings").appendChild(el);
+    }
+
+    $("#result").classList.remove("hidden");
+  },
+
+  hideSubmission: function() {
+    $("#submit-form").classList.add("hidden");
+  },
+
+  showSubmission: function() {
+    $("#checkbox-owner").checked = false;
+    $("#checkbox-subdomains").checked = false;
+    $("#submit").disabled = true;
+
+    $("#submit-form").classList.remove("hidden");
+  },
+
+  onSubmitForm: function() {
+    this.submit(this.domainToSubmit).then(function(issues) {
+      console.log("submit:", issues);
+      $("#submit-result").classList.remove("hidden");
+      if (issues.errors.length == 0) {
+        $("#submit-result").textContent = "Submitted successfully!";
+        // TODO: Now try SSL Labs!
+      } else {
+        $("#submit-result").textContent = "There are errors. Please submit your site again.";
+      }
+    });
+  },
+
+  showWaiting: function(domain) {
+    $("#checking").textContent = "Checking  " + domain;
+
+    $("#result-waiting").classList.remove("hidden");
+  },
+
+  hideWaiting: function() {
+    $("#result-waiting").classList.add("hidden");
+  },
+
+  updateURLHash: function(domain) {
+    console.log(domain);
+    if (domain) {
+      history.replaceState({}, document.title, ".?domain=" + domain);
+    } else {
+      history.replaceState({}, document.title, ".");
```

Comment:
Hmm, I'd normally be in favor. However, the "." argument is so unusual that I strongly prefer to isolate each case explicitly here.